libraries/sharedpref-manager/src/main/java/com/trendyol/android/devtools/sharedprefmanager/domain/SharedPrefUpdateTypeValidator.kt

```diff
@@ -0,0 +1,65 @@
+package com.trendyol.android.devtools.sharedprefmanager.domain
+
+import com.trendyol.android.devtools.sharedprefmanager.data.SharedPrefModel
+import com.trendyol.android.devtools.sharedprefmanager.data.SharedPrefValueType
+
+class SharedPrefUpdateTypeValidator {
+
+    fun validate(sharedPrefModel: SharedPrefModel, newValue: String) {
+        when (sharedPrefModel.valueType) {
+            SharedPrefValueType.INT -> {
+                kotlin.runCatching {
+                    newValue.toInt()
+                }.onSuccess {
```

Comment:
Empty `onSuccess` blocks can be removed and exceptions can be gathered in a function like below:

```kotlin
fun throwExceptionFor(valueType: SharedPrefValueType) {
    throw IllegalArgumentException("Old value type is: $valueType so new value type must be $valueType.")
}
```

Reply:
done with 8b14e3d43cf31640a2c0a5110f5988b05c794660